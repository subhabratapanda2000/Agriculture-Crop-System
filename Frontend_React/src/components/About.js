import React from "react";

import img1 from "../images/about_us_crop.png";
import Header from "./Header";
import { Link } from "react-router-dom";

const About = () => {
  return (
    <div>
      <Header />
      <div
        style={{
          backgroundImage: `url(${img1})`,
          height: "700px",
          width: "100%",
          backgroundRepeat: "no-repeat",
          backgroundSize: "cover",
          backgroundBlendMode: "lighten",
        }}
      >
        <p
          className="display-1 text-center pb-1"
          style={{
            paddingTop: "40px",
            paddingLeft: "40px",
            paddingRight: "40px",
            fontSize: "24px",
            border: "2px solid ",
            borderRadius: "25px",
            backgroundColor: "rgba(216, 227, 225, 0.5)",
            height: "220px",
          }}
        >
          This <b>Agriculture Crop System</b> act as a bridge between the Farmer
          and the Dealer (Crop purchaser). There is no need for the farmer to
          carry the crop till the market, paying unexpected commissions and wait
          for proper price from the Dealers. Whenever farmer want to sell the
          vegetables/fruits at the Farm itself, he just wants to select the type
          of Crop, quantity available, input the location/address and publish
          the information to Dealers. Whoever(Dealer) is interested in
          purchasing will connect with the Farmer, will reach the location,
          checks the quality of the crop, negotiates the price and buy the crop.
        </p>
      </div>
    </div>
  );
};

export default About;
