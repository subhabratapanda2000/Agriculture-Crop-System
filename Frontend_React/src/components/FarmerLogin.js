import React from "react";
import "../style/farmerlogin.css";
import { useState, useEffect } from "react";
import { Link, useHistory, useParams } from "react-router-dom";
import img1 from "../images/farmer-icon.png";
import farmerService from "../services/auth.service";
import img3 from "../images/farmer_field2.jpg";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import Header from "./Header";

const FarmerLogin = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const history = useHistory();
  const [loading, setLoading] = useState(false);

  const [usernameValid, setUsernameValid] = useState(false);
  const [passwordValid, setPasswordValid] = useState(false);

  const [usernameTouched, setUsernameTouched] = useState(false);
  const [passwordTouched, setPasswordTouched] = useState(false);
  const [errAlert, setErrAlert] = useState(false);

  if (farmerService.getCurrentUser()) {
    const user = farmerService.getCurrentUser();

    if (user.role === "ROLE_FARMER") {
      history.push("/farmerDashboard");
    } else if (user.role === "ROLE_DEALER") {
      history.push("/dealerDashboard");
    } else if (user.role === "ROLE_ADMIN") {
      history.push("/adminDashboard");
    } else {
      history.push("/error");
    }
  }

  //handler
  const usernameHandler = (e) => {
    setUsername(e.target.value);
    if (e.target.value.trim() !== "") {
      setUsernameValid(true);
    }
  };

  const passwordHandler = (e) => {
    setPassword(e.target.value);
    if (e.target.value.trim() !== "") {
      setPasswordValid(true);
    }
  };

  // blur handler
  const usernameBlurHandler = (e) => {
    if (username.trim() == "") {
      setUsernameTouched(true);
      setUsernameValid(false);
      return;
    }
  };

  const passwordBlurHandler = (e) => {
    if (password.trim() == "") {
      setPasswordTouched(true);
      setPasswordValid(false);
      return;
    }
  };

  const save = () => {
    history.push("/addFarmer");
  };

  const saveFarmer = (e) => {
    setLoading(true);
    setErrAlert(false);
    setUsernameTouched(true);
    setPasswordTouched(true);

    if (username.trim() === "") {
      setUsernameValid(false);
      setLoading(false);
      return;
    }
    setUsernameValid(true);

    if (password.trim() === "") {
      setLoading(false);
      setPasswordValid(false);
      return;
    }
    setPasswordValid(true);
    farmerService
      .login(username, password)
      .then((response) => {
        const user = farmerService.getCurrentUser();
        console.log(user.jwt);
        console.log(user.id);
        toast.success("Login Sussessful");
        if (user.role == "ROLE_FARMER") {
          history.push(`/farmerDashboard`);
          // window.location.reload(false);
        } else {
          history.push(`/error`);
          toast.error("You are not a Farmer");
        }
      })
      .catch((error) => {
        setErrAlert(true);
        setLoading(false);
        toast.error("Wrong Username or Password");

        // alert("Wrong UserName or Password");
      });
    console.log(username, password);
  };

  const addAlert = "Wrong UserName or Password";
  const usernameInputValid = !usernameValid && usernameTouched;
  const passwordInputValid = !passwordValid && passwordTouched;

  const usernameStyle = usernameInputValid
    ? " form-control col-8 border border-3 border-danger"
    : "";
  const passwordStyle = passwordInputValid
    ? " form-control col-8 border border-3 border-danger"
    : "";

  const sty = ``;
  return (
    <div>
      <Header />

      <div
        style={{
          backgroundImage: `url(${img3})`,
          backgroundAttachment: "fixed",
          height: "720px",
          width: "100%",
          backgroundRepeat: "no-repeat",
          backgroundSize: "cover",
          backgroundBlendMode: "lighten",
          paddingTop: "5px",
        }}
      >
        <div
          id="d1"
          style={{
            paddingTop: "5px",
            border: "solid",
            height: "85%",
            border: "2px solid ",
            borderRadius: "25px",
            backgroundColor: "rgba(216, 227, 225, 0.7)",
          }}
        >
          {errAlert && (
            <div
              class="alert alert-warning alert-dismissible fade show"
              role="alert"
              style={{
                backgroundColor: "red",
                color: "white",
                marginTop: "15px",
              }}
            >
              <strong>{addAlert}</strong>
              <button
                type="button"
                class="btn-close"
                data-bs-dismiss="alert"
                aria-label="Close"
                onClick={() => {
                  setErrAlert(false);
                }}
              ></button>
            </div>
          )}
          <img src={img1} className="img" path="/admin/login" />
          <p className="p7">Farmer LOGIN</p>
          <input
            className={usernameStyle}
            type="text"
            id="username"
            value={username}
            // onChange={(e) => setUsername(e.target.value)}
            onChange={usernameHandler}
            onBlur={usernameBlurHandler}
            placeholder="USERNAME"
            required
          />
          {usernameInputValid && (
            <p className="text-danger text-center font-weight-bold">
              Username must not be empty
            </p>
          )}
          <br />
          <br />
          <p>
            <input
              className={passwordStyle}
              type="password"
              id="password"
              value={password}
              // onChange={(e) => setPassword(e.target.value)}
              onChange={passwordHandler}
              onBlur={passwordBlurHandler}
              placeholder="PASSWORD"
              required
            />
          </p>
          {passwordInputValid && (
            <p className="text-danger text-center font-weight-bold">
              Password must not be empty
            </p>
          )}
          <br />

          <input
            type="submit"
            onClick={(e) => saveFarmer(e)}
            value="Login Now"
          />

          {/*  <p1>Remeber me </p1> */}
          <p
            style={{
              textAlign: "center",
              paddingLeft: "10px",
              paddingTop: "10px",
              color: "black",
            }}
          >
            New Farmer?
            <Link onClick={() => save()}> &nbsp; &nbsp; Sign Up</Link>
          </p>
          <div className="d-flex justify-content-center">
            {loading && (
              <span
                style={{ width: "3rem", height: "3rem" }}
                className="spinner-border spinner-border-large text-primary"
              ></span>
            )}
          </div>
          {/*  <br><br><br>
   <p1><center>not a member?</center></p1>
   <button>Create account</button>  */}
        </div>

        {/* <marquee direction="right">
        <div>
          <img
            src={img5}
            style={{ width: "140px", height: "120px", paddingBottom: "10px" }}
          />
        </div>
      </marquee> */}
      </div>
    </div>
  );
};

export default FarmerLogin;
