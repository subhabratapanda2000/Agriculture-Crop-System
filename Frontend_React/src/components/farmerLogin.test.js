// import React from "react";
// import ReactDOM from "react-dom";
// import FarmerLogin from "./FarmerLogin";
// import { render, screen } from "@testing-library/react";

// // it("renders without crashing", () => {
// //   const div = document.createElement("div");
// //   ReactDOM.render(<FarmerLogin></FarmerLogin>, div);
// // });

// test("renders learn react link", () => {
//   render(<FarmerLogin />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

import React from "react";
import ReactDOM from "react-dom";
import FarmerLogin from "./FarmerLogin";
import About from "./About";

import { render } from "@testing-library/react";
// import "jest-dom/extend-expect";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

it("matches snapshot", () => {
  const tree = renderer.create(<About></About>).toJSON();
  console.log("Farmer login");
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
