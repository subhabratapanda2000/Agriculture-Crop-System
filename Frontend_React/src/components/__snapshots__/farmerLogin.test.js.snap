// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundBlendMode": "lighten",
      "backgroundImage": "url(about_us_crop.png)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "height": "700px",
      "width": "100%",
    }
  }
>
  <p
    className="display-1 text-center pb-1"
    style={
      Object {
        "backgroundColor": "rgba(216, 227, 225, 0.5)",
        "border": "2px solid ",
        "borderRadius": "25px",
        "fontSize": "24px",
        "height": "220px",
        "paddingLeft": "40px",
        "paddingRight": "40px",
        "paddingTop": "40px",
      }
    }
  >
    This 
    <b>
      Agriculture Crop System
    </b>
     act as a bridge between the Farmer and the Dealer (Crop purchaser). There is no need for the farmer to carry the crop till the market, paying unexpected commissions and wait for proper price from the Dealers. Whenever farmer want to sell the vegetables/fruits at the Farm itself, he just wants to select the type of Crop, quantity available, input the location/address and publish the information to Dealers. Whoever(Dealer) is interested in purchasing will connect with the Farmer, will reach the location, checks the quality of the crop, negotiates the price and buy the crop.
  </p>
</div>
`;
